src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/AbstractBuildEventValueProvider.AbstractBuildEventHelper.cs

```diff
@@ -0,0 +1,192 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Construction;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties.InterceptedProjectProperties
+{
+    internal abstract partial class AbstractBuildEventValueProvider
+    {
+        public abstract class AbstractBuildEventHelper
+        {
+            private const string ExecTask = "Exec";
+            private const string Command = "Command";
+
+            protected AbstractBuildEventHelper(string buildEvent,
+                   string targetName,
+                   Func<ProjectTargetElement, string> getTarget,
+                   Action<ProjectTargetElement> setTargetDependencies)
+            {
+                BuildEvent = buildEvent;
+                TargetName = targetName;
+                GetTarget = getTarget;
+                SetTargetDependencies = setTargetDependencies;
+            }
+
+            private Func<ProjectTargetElement, string> GetTarget { get; }
+            private Action<ProjectTargetElement> SetTargetDependencies { get; }
+            private string BuildEvent { get; }
+            private string TargetName { get; }
+
+            public string GetProperty(ProjectRootElement projectXml)
+            {
+                // Check if project file already has props in place for this.
+                var result = TryGetFromProperty(projectXml);
+                if (result.success)
+                {
+                    return result.property;
+                }
+
+                // Check if build events can be found in targets
+                return GetFromTargets(projectXml);
+            }
+
+            public void SetProperty(string unevaluatedPropertyValue, ProjectRootElement projectXml)
+            {
+                // Check if project file already has props in place for this.
+                if (TrySetProperty(unevaluatedPropertyValue, projectXml))
+                {
+                    return;
+                }
+
+                if (OnlyWhitespaceCharacters(unevaluatedPropertyValue))
+                {
+                    var result = FindTargetToRemove(projectXml);
+                    if (result.success)
+                    {
+                        projectXml.RemoveChild(result.target);
+                        return;
+                    }
+                }
+
+                SetParameter(projectXml, unevaluatedPropertyValue);
+            }
+
+            private (bool success, string property) TryGetFromProperty(ProjectRootElement projectXml)
+            {
+                // Choose last or default as that matches the evaluation order for MSBuild
+                var property = projectXml.Properties
```

Comment:
Ah actually just noticed that you are implementing an intercepter for this property anyway. So just check if evaluatedProeprtyValue [here](https://github.com/dotnet/project-system/pull/2367/files#diff-df8f51b25a07aeabf5c8cb6e9b93f764R26) is non-null. You can avoid this whole TryGet\SetFromProperty thing. You'd still need to change SourceOfDefaultValue.